core/src/mast/node/basic_block_node/mod.rs

```diff
@@ -231,7 +296,7 @@ impl PrettyPrint for BasicBlockNode {
     }
 }
 
-impl fmt::Display for BasicBlockNode {
+impl<'a> fmt::Display for BasicBlockNodePrettyPrint<'a> {
```

Comment:
FYI `Display` is implemented for `dyn PrettyPrint`, so a reference to any `PrettyPrint` impl automatically implements `Display`. Can still be convenient to impl `Display` explicitly, like done here, but figured it's useful info to have on hand anyway.